### PR TITLE
Implement AccessFee v2

### PR DESCRIPTION
**Pull request by ViktorPetrovski**

### CODE REVIEW

File: Data/src/main/java/com/sdk/data/model/asset/AccessFeeModel.kt [hunk 1]
```diff
@@ -6,11 +6,18 @@ data class AccessFeeModel(
     @SerializedName("id") val id: Long,
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/AccessFeeModel.kt:7]: Made them optional

File: Data/src/main/java/com/sdk/data/model/asset/AccessTypeModel.kt [hunk 2]
```diff
@@ -1,4 +1,9 @@
 package com.sdk.data.model.asset
 
 
-data class AccessTypeModel(val id: Long, val name: String, val quantity: Int, val period: String)
\ No newline at end of file
+data class AccessTypeModel(
+    val id: Long,
+    val account_id: Long,
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/AccessTypeModel.kt:8]: Made them optional

File: Data/src/main/java/com/sdk/data/model/asset/ExternalFeesApiModel.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.sdk.data.model.asset
+
+import com.google.gson.annotations.SerializedName
+
+data class ExternalFeesApiModel(
+    @SerializedName("id") val id: Long?,
+    @SerializedName("payment_provider_id") val paymentProviderId: Long?,
+    @SerializedName("access_fee_id") val accessFeeId: Long?,
+    @SerializedName("external_id") val externalId: String?,
+    @SerializedName("merchant_id") val merchantId: Long
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/ExternalFeesApiModel.kt:11]: Optional

File: Data/src/main/java/com/sdk/data/model/asset/GeoRestrictionApiModel.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.sdk.data.model.asset
+
+data class GeoRestrictionApiModel(
+    val id: Long,
+    val country_iso: String?,
+    val country_set_id: Long,
+    val type: String?
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/GeoRestrictionApiModel.kt:9]: Made them optional

File: Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+data class VoucherApiModel(
+    val code: String?,
+    val discount: Int,
+    val discount_duration: Int,
+    val discount_period: String?,
+    val end_date: String?,
+    val id: Int,
+    val name: String?,
+    val rebill_discount: Int,
+    val start_date: String?,
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt]: It's not String, should be Double/Long

File: Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package com.sdk.data.model.asset.v2
+
+import com.sdk.domain.entity.asset.VoucherEntity
+
+data class VoucherApiModel(
+    val code: String?,
+    val discount: Int,
+    val discount_duration: Int,
+    val discount_period: String?,
+    val end_date: String?,
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt]: It's not String, it's Double/Long

File: Data/src/main/java/com/sdk/data/remote/AssetsRemoteImpl.kt [hunk 7]
```diff
@@ -37,4 +38,7 @@ class AssetsRemoteImpl constructor(
         return inPlayerRemotePublicServiceAPI.getAccessFees(id = id)
     }
     
+    override fun getAccessFeesV2(id: Int, voucher: Int): Single<List<AccessFeeModelV2>> {
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/remote/AssetsRemoteImpl.kt]: Voucher should be optional value.

File: Data/src/main/java/com/sdk/data/repository/InPlayerAssetsRepositoryImpl.kt [hunk 8]
```diff
@@ -36,4 +36,10 @@ class InPlayerAssetsRepositoryImpl(private val assetsRemote: AssetsRemote,
             list.map { mapAccessFee.mapFromModel(it) }
         }
     }
+    
+    override fun getAccessFeesv2(id: Int, voucher: Int): Single<List<AccessFeeEntity>> {
```
Comment (odimitrov) [inline Data/src/main/java/com/sdk/data/repository/InPlayerAssetsRepositoryImpl.kt]: Same here, voucher should be optional value

File: Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt (hunk 6 above)
Comment (ViktorPetrovski) [reply Data/src/main/java/com/sdk/data/model/asset/v2/VoucherApiModel.kt]: Changed

File: Data/src/main/java/com/sdk/data/repository/InPlayerAssetsRepositoryImpl.kt (hunk 8 above)
Comment (ViktorPetrovski) [reply Data/src/main/java/com/sdk/data/repository/InPlayerAssetsRepositoryImpl.kt]: Updated in newest commit

**Review by odimitrov — DISMISSED** [review]

**Review by wbmd-odimitrov — APPROVED** [review]

**Review by wbmd-odimitrov — APPROVED** [review]

**Review by odimitrov — APPROVED** [review]

**Pull request merged**